### PR TITLE
[GCI-266] React UI Library - Button component

### PR DESCRIPTION
**Pull request by TomaszDomagala**
#### issue: https://issues.openmrs.org/browse/GCI-266
Button props: 

- type (submit, cancel)

- isFlat (boolean) if true button has no background and shadow

- isDisabled (boolean) if true button has no hover and can't be clicked

- onClick (button behavior when clicked)

### CODE REVIEW

File: src/components/Obutton/index.jsx [hunk 1]
```diff
@@ -0,0 +1,43 @@
+import React, {Component} from 'react';
+import RaisedButton from 'material-ui/RaisedButton';
+import FlatButton from 'material-ui/FlatButton';
+
+class Obutton extends Component {
```
Comment (merovingienne) [inline src/components/Obutton/index.jsx]: Let's rename the component to Button. We can overlook possible future conflicts right now.

File: src/components/Obutton/index.jsx [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    isParamTrue(param) {
+        return param === true || param === "true";
+    }
+
+    render() {
+        let disabled = this.props.disabled;
+        let flat = this.props.flat;
+        let onClick = this.props.onClick;
+        let backgroundColor = "#ffffff";
```
Comment (merovingienne) [inline src/components/Obutton/index.jsx]: Refactor these to separate style objects. Eg: flatButtonStyle

File: src/components/Obutton/index.jsx [hunk 3]
```diff
@@ -0,0 +1,43 @@
+import React, {Component} from 'react';
+import RaisedButton from 'material-ui/RaisedButton';
+import FlatButton from 'material-ui/FlatButton';
+
+class Obutton extends Component {
+    isParamTrue(param) {
+        return param === true || param === "true";
```
Comment (merovingienne) [inline src/components/Obutton/index.jsx]: 3. This is not necessary, the developer must provide a boolean prop.

File: src/components/Obutton/index.jsx [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    }
+
+    render() {
+        let disabled = this.props.disabled;
```
Comment (merovingienne) [inline src/components/Obutton/index.jsx]: 4. No need to do this. Refer 3.

File: src/components/Obutton/index.jsx [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+                flatFontColor = "#00C853";
+                raisedFontColor = "#ffffff";
+                break;
+            case "cancel":
+                backgroundColor = "#D50000";
+                flatFontColor = "#D50000";
+                raisedFontColor = "#ffffff";
+                break;
+        }
+        if (this.isParamTrue(disabled) && this.isParamTrue(flat))
```
Comment (merovingienne) [inline src/components/Obutton/index.jsx]: Directly check props. Refer 3 & 4. And refactor the render logic to a separate method and call that here. Also add default props and mention prop types. See https://reactjs.org/docs/typechecking-with-proptypes.html.

**Comment by TomaszDomagala** [comment]
@merovingienne I think I've done everything

**Review by BartlomiejRasztabiga — APPROVED** [review]

**Pull request merged**